sound/soc/sof/intel/hda-loader.c

```diff
@@ -398,23 +427,7 @@ int hda_dsp_cl_boot_firmware(struct snd_sof_dev *sdev)
 		goto cleanup;
 	}
 
-	/*
-	 * When a SoundWire link is in clock stop state, a Slave
-	 * device may trigger in-band wakes for events such as jack
-	 * insertion or acoustic event detection. This event will lead
-	 * to a WAKEEN interrupt, handled by the PCI device and routed
-	 * to PME if the PCI device is in D3. The resume function in
-	 * audio PCI driver will be invoked by ACPI for PME event and
-	 * initialize the device and process WAKEEN interrupt.
-	 *
-	 * The WAKEEN interrupt should be processed ASAP to prevent an
-	 * interrupt flood, otherwise other interrupts, such IPC,
-	 * cannot work normally.  The WAKEEN is handled after the ROM
-	 * is initialized successfully, which ensures power rails are
-	 * enabled before accessing the SoundWire SHIM registers
-	 */
-	if (!sdev->first_boot)
-		hda_sdw_process_wakeen(sdev);
+	hda_sdw_resume(sdev);
```

Comment:
we could move the !sdev->first_boot inside the helper, no need to add a new definition that's rather misleading.

Reply:
> we could move the !sdev->first_boot inside the helper, no need to add a new definition that's rather misleading.

checking sdev->first_boot inside hda_sdw_process_wakeen() sounds a bit odd to me, @plbossart do you really want that?

Reply:
I actually don't see a need for a new wrapper here, just use 	

if (!sdev->first_boot)
		hda_sdw_process_wakeen(sdev);

twice...

Reply:
actually the imr stuff only happens after the first boot so the test is not needed ... really no need for a wrapper.